crates/turborepo-lib/src/commands/mod.rs

```diff
@@ -1,2 +1,40 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+
+use crate::{
+    config::{
+        default_user_config_path, RepoConfig, RepoConfigLoader, UserConfig, UserConfigLoader,
+    },
+    ui::UI,
+    Args,
+};
+
 pub(crate) mod bin;
 pub(crate) mod logout;
+
+pub struct CommandBase {
+    pub repo_root: PathBuf,
+    pub ui: UI,
+    pub repo_config: RepoConfig,
+    pub user_config: UserConfig,
+}
+
+impl CommandBase {
+    pub fn new(args: Args, repo_root: PathBuf) -> Result<Self> {
+        let repo_config_path = repo_root.join(".turbo").join("config.json");
```

Comment:
This operations should probably live in `config/repo.rs` for better discoverability